taskmgr/server/server_test.go

```diff
@@ -39,3 +42,58 @@ func TestGetTasksHandler(t *testing.T) {
 	})
 
 }
+
+func TestNewTaskHandler(t *testing.T) {
+
+	var IMD = inMemoryDatabase{
+		tasks: []*task.Task{},
+	}
+
+	newDummyTask := struct {
+		Title string
+		Desc  string
+	}{
+		Title: "new upcomig task",
+		Desc:  "title of a new fancy task",
+	}
+
+	t.Run("check new task created", func(t *testing.T) {
+
+		jsonTask, err := json.Marshal(newDummyTask)
+		if err != nil {
+			t.Errorf("Unable to marshal the dummy task")
+			return
+		}
+
+		request, _ := http.NewRequest(http.MethodPost, "/new", bytes.NewBuffer(jsonTask))
+
+		body, err := io.ReadAll(request.Body)
+		if err != nil {
+			t.Errorf("Couldn't read the body. Error type: %s", err)
+		}
+		defer request.Body.Close()
+
+		newTask := task.Task{
+			ID: uuid.New(),
+		}
+
+		err = json.Unmarshal(body, &newTask)
+		if err != nil {
+			t.Errorf("Invalid JSON format")
+			return
+		}
+		IMD.New(&newTask)
+
+		if len(IMD.tasks) == 0 {
+			t.Errorf("Expected a task to be added, but the task list is empty")
+		} else {
+			got := IMD.tasks[0].Title
+			want := newDummyTask.Title
+
+			if got != want {
+				t.Errorf("got %q, want %q", got, want)
+			}
+		}
+
+	})
```

Comment:
try to avoid nesting code as much as you can, if else usually can become just if by inverting the check we are making
```suggestion
		if len(IMD.tasks) == 0 {
			t.Fatalf("Expected a task to be added, but the task list is empty")
		} 
		
		got := IMD.tasks[0]
	        want := newDummyTask

                 // reflect.DeepEqual compares structs and all their fields so you don't have to chose one to check you can check the whole thing is equal.
		if !reflect.DeepEqual(got, want) {
			t.Errorf("got %v, want %v", got, want)
		}
	}

	})
```